apps_test.go

```diff
@@ -0,0 +1,134 @@
+package sonobuoy_plugin
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	appv1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/ctrlclient"
+)
+
+// Test_Apps test the default release apps are installed and deployed successfully.
+func Test_Apps(t *testing.T) {
+	t.Parallel()
+
+	var err error
+
+	ctx := context.Background()
+
+	cpCtrlClient, err := ctrlclient.CreateCPCtrlClient()
+	if err != nil {
+		t.Fatalf("error creating CP k8s client: %v", err)
+	}
+
+	regularLogger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	logger := NewTestLogger(regularLogger, t)
+
+	clusterID, exists := os.LookupEnv("CLUSTER_ID")
+	if !exists {
+		t.Fatal("missing CLUSTER_ID environment variable")
+	}
+
+	logger.Debugf(ctx, "Testing that all apps declared in the Release CR are deployed.")
+
+	// Get Cluster.
+	logger.Debugf(ctx, "Getting cluster CR.")
+	var cluster capiv1alpha3.Cluster
+	{
+		clusters := &capiv1alpha3.ClusterList{}
+		err = cpCtrlClient.List(ctx, clusters, client.MatchingLabels{label.Cluster: clusterID})
```

Comment:
I mean, this works, but if we have the actual `CLUSTER_ID` wouldn't it make more sense just using `Get` it instead of using `List` and making sure there is only 1 item?

Reply:
yes it makes sense let me do it

Reply:
but wait, what about the namespace?

Reply:
oooh you are right, we don't know the organization, so we can't do it like I suggested. Anyway, I remembered that we have some utility functions for this in `pkg/capiutil/cluster.go`.